packages/web/hooks/limit-orders/use-place-limit.ts

```diff
@@ -58,59 +58,87 @@ export const usePlaceLimit = ({
   });
   const account = accountStore.getWallet(osmosisChainId);
 
-  const paymentAmount = useMemo(
-    () =>
-      orderDirection === OrderDirection.Ask
-        ? inAmountInput.amount ?? new CoinPretty(baseAsset, "0")
-        : new CoinPretty(
-            quoteAsset,
-            inAmountInput.amount?.toCoin().amount ?? "0"
-          ).mul(priceState.price),
-    [
-      orderDirection,
-      inAmountInput.amount,
-      priceState.price,
-      baseAsset,
-      quoteAsset,
-    ]
-  );
-
   const { price: baseAssetPrice } = useCoinPrice(
     new CoinPretty(baseAsset, new Dec(1))
   );
   const { price: quoteAssetPrice } = useCoinPrice(
     new CoinPretty(quoteAsset, new Dec(1))
   );
 
-  const paymentFiatValue = useMemo(
-    () =>
-      mulPrice(
-        paymentAmount,
-        orderDirection === OrderDirection.Bid
-          ? quoteAssetPrice
-          : baseAssetPrice,
-        DEFAULT_VS_CURRENCY
-      ),
-    [paymentAmount, orderDirection, baseAssetPrice, quoteAssetPrice]
-  );
+  /**
+   * Calculates the amount of tokens to be sent with the order.
+   * In the case of an Ask order the amount sent is the amount of tokens defined by the user in terms of the base asset.
+   * In the case of a Bid order the amount sent is the requested fiat amount divided by the current quote asset price.
+   * The amount is then multiplied by the number of decimal places the quote asset has.
+   *
+   * @returns The amount of tokens to be sent with the order in base asset amounts for an Ask and quote asset amounts for a Bid.
+   */
+  const paymentTokenValue = useMemo(() => {
+    // The amount of tokens the user wishes to buy/sell
+    const baseTokenAmount =
+      inAmountInput.amount ?? new CoinPretty(baseAsset, new Dec(0));
+    if (orderDirection === OrderDirection.Ask) {
+      // In the case of an Ask we just return the amount requested to sell
+      return baseTokenAmount;
+    } else {
```

Comment:
note for readability: is this else block necessary if you are guard-clausing with the if -> return in the condition before?

Reply:
The else block isn't necessary, was just how I wrote it as I thought through the logic. Will remove it.